da/celestia/celestia.go

```diff
@@ -111,34 +134,56 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *ty
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultCheckBlock {
-	shares, err := c.client.NamespacedShares(ctx, c.namespaceID, dataLayerHeight)
-	code := dataRequestErrorToStatus(err)
-	if code != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
+	header, err := c.rpc.Header.GetByHeight(ctx, dataLayerHeight)
+	if err != nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	if header.DAH == nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:     da.StatusSuccess,
+				DAHeight: dataLayerHeight,
+			},
+			DataAvailable: false,
+		}
+	}
+	err = c.rpc.Share.SharesAvailable(ctx, header.DAH)
+	if err != nil {
+		if strings.Contains(err.Error(), ErrNotAvailable.Error()) {
+			return da.ResultCheckBlock{
+				BaseResult: da.BaseResult{
+					Code:     da.StatusSuccess,
+					DAHeight: dataLayerHeight,
+				},
+				DataAvailable: false,
+			}
+		}
 		return da.ResultCheckBlock{
 			BaseResult: da.BaseResult{
 				Code:    da.StatusError,
 				Message: err.Error(),
 			},
 		}
 	}
-
 	return da.ResultCheckBlock{
 		BaseResult: da.BaseResult{
 			Code:     da.StatusSuccess,
 			DAHeight: dataLayerHeight,
 		},
-		DataAvailable: len(shares) > 0,
+		DataAvailable: true,
 	}
 }
 
 // RetrieveBlocks gets a batch of blocks from DA layer.
-func (c *DataAvailabilityLayerClient) RetrieveBlocks(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultRetrieveBlocks {
-	data, err := c.client.NamespacedData(ctx, c.namespaceID, dataLayerHeight)
+func (c *DataAvailabilityLayerClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64) da.ResultRetrieveBlocks {
+	c.logger.Debug("querying GetAll blobs with params", "height", dataLayerHeight, "namespace", hex.EncodeToString(c.namespace.Bytes()))
+	blobs, err := c.rpc.Blob.GetAll(ctx, dataLayerHeight, []namespace.ID{c.namespace.Bytes()})
```

Comment:
why convert to `c.namespace.Bytes()`? can you use `c.namespace.ID`? it should fix the `ErrNamespaceNotFound` problem?

Reply:
because `Bytes()` is going to append the `c.namespace.Version`, which is probably the reason for namespace not found error.

Reply:
updated, thanks.

Reply:
Using `namespace.Bytes`, the `namespace not found` error is only happening on heights without data, it's working fine when retrieving blocks.

Reply:
i see. then it should be fine as the namespace not found is considered success status.